### DISCUSSION TITLE
USD Camera and USD Nurbs Curve Support

### DISCUSSION THREAD

**Discussion by brycegbrazen** [Autodesk/maya-usd#2022] [Q&A] [answered]
Hi all,

Test environment:
Maya 2022.3
MayaUSD 0.15.0

Firstly, just curious if someone has any information on support of UsdGeomCamera in Maya. A few things that I have noticed at a quick glance is that the clipping planes are not easily set in the Attribute Editor, and that the camera has no physical representation in viewport. A breakdown of Maya support of the USD camera vs the standard Maya camera would be great to know.
![image](https://user-images.githubusercontent.com/69012021/150861815-3d244458-4c39-4602-8d8f-46c6366da0a2.png)

Secondly, I wanted to ask about the current state of UsdGeomNurbsCurves in Maya. I have exported a very basic NURBS curve from Maya via Export Selected menu item. When loading a curve USD file as USD stage or as a reference, I am seeing an issue where the curve is not rendered in the viewport. When selecting the curve in the Outliner, it appears correctly in the viewport. This seems like a bug, but I'm not sure if anyone else can replicate this issue. I'll attach a basic USD file to reproduce the issue.
[usd_curve_export_test.zip](https://github.com/Autodesk/maya-usd/files/7928797/usd_curve_export_test.zip)

Thanks for the help,
Bryce G

**Comment by kxl-adsk** [comment, marked as answer]
Hi @brycegbrazen,

The AE has still limited support for different data types and we fallback to generic support which operates on strings and doesn't allow for editing. This is something that can be improved from the plugin side leveraging the work we did on supporting metadata via custom controls. Our end goal here is to add all the missing data types to UFE.

Regarding nurbs curves support. I wonder if you are not running into issue with color where both background and curves had exactly the same color. We addressed it recently with this PR - https://github.com/Autodesk/maya-usd/pull/1998

Cheers,
KxL

**Reply by brycegbrazen** [reply]
Hi @kxl-adsk,

Thanks for the info! One other question about this, it seems the USD camera has no physical representation in the viewport. Is this something you guys are aware of as well?

Thanks,
Bryce

**Reply by neilrubie** [reply]
Unfortunately this doesn't seem to be a priority:
https://github.com/Autodesk/maya-usd/issues/1451#issuecomment-1009179956

**Reply by kxl-adsk** [reply]
Wasn't a priority for 0.16.0 and won't be for 0.17.0...but it is on our radar. It is a bigger feature because it's going to involve changes on Maya's side. At the same time, we will want to address manipulation when looking through the camera.

**Reply by brycegbrazen** [reply]
Just noting this here, regarding the NURBS Curve support, the issue is fixed for the example scene I provided in mayaUSD v0.16.0. So your above statement was correct:

```Regarding nurbs curves support. I wonder if you are not running into issue with color where both background and curves had exactly the same color. We addressed it recently with this PR - https://github.com/Autodesk/maya-usd/pull/1998```

**Reply by brycegbrazen** [reply]
@kxl-adsk Any update on the addition of USD camera support in the viewport?

**Reply by kxl-adsk** [reply]
Hi @brycegbrazen. You will have to wait for someone from MayaUSD to respond. I no longer work at Autodesk. Cheers!

**Reply by santosd** [reply]
Hi Krystian 👋. @brycegbrazen there have been some changes regarding UFE cameras integrated however, some of the changes require changes in Maya and therefore we are hoping to get these out in a future update of Maya.

**Reply by brycegbrazen** [reply]
@santosd As of Maya 2023.2, I am now able to see the USD camera in the viewport. Does this line up with your expectations? Also, is there an ETA on when we will be able to natively change complex USD values in Maya such as the Clipping Range on the camera?
![image](https://user-images.githubusercontent.com/69012021/201745364-bc74089c-c606-467d-a0df-0bde3d44cb0c.png)

**Reply by santosd** [reply]
Hi @brycegbrazen , Being able to see the USD camera in the viewport as of 2023.2 does make sense. In 2023.1 and Maya USD 0.18.0 we added support for UFE cameras to draw in the Viewport. In regards to the changing complex USD values such as the Clipping Range on the camera it seems that those changes require Maya side changes so they should be coming in a future Maya release.
